### PR TITLE
feat(connectors): fixes APPSRE-4651, added support for vault-secret-prefix flag to add a prefix to all aws vault secret keys, defaults to managed-connectors/

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Added support for vault-secret-prefix flag to add a prefix to al aws vault secret keys, defaults to managed-connectors/

## Verification Steps
Included unit test can be run with valid aws credentials file. 

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
~~- [ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [x] Code Review completed
~~- [ ] Verified independently by reviewer~~
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

**Comment by heiko-braun** [comment]
@dhirajsb Can you hold this change for now until we had a chance to discuss the ramifications?

**Comment by BumbleFeng** [comment]
I suggest making SecretPrefix optional, so we can separate promoting code with migrating secrets in two steps.

**Comment by dhirajsb** [comment]
@BumbleFeng I added a flag to enable/disable secret prefix. I'll also add that to the cos-fleet-manager templates once this PR is merged. 
@lburgazzoli we need to get this merged for sre acceptance.

File: internal/connector/internal/services/vault/vault_service_aws.go
Comment (valdar) [inline internal/connector/internal/services/vault/vault_service_aws.go:20]: Is an enabling boolean necessary?
After all if you specify a prefix you implicitly enable the feature right? What is the point of specifying a prefix and disabling the feature?

**Review by valdar — CHANGES_REQUESTED** [review]

File: internal/connector/internal/services/vault/vault_service_aws.go
Comment (dhirajsb) [reply internal/connector/internal/services/vault/vault_service_aws.go:20]: Yes, it's necessary if one also wants to support a default value for a prefix. Otherwise it's a nightmare to figure out whether it's enabled or not by having to look at every point of configuration i.e, code, template, and app-services config. With a switch in the app-services config, it will be self documenting.

**Review by valdar — APPROVED** [review]

**Pull request merged**
